spec/features/admin/admin_can_see_backend_spec.rb

```diff
@@ -0,0 +1,36 @@
+require 'rails_helper'
+
+describe 'Authorization' do
+  describe 'an admin' do
+    scenario 'can access the admin backend' do
+      user = User.create!(
+        name: 'Bob Ross',
+        email: 'bob@bobross.com',
+        password: 'happy_little_trees',
+        role: 1
+      )
+
+      allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(user)
+
+      visit admin_root_path
+      expect(page).to have_content('Travelistics Admin')
+    end
+  end
+
+  describe 'a regular user' do
+    scenario 'cannot access admin backend' do
```

Comment:
@iandouglas @AliSchlereth This is the only location in which I'm testing this functionality... should I be adding a test for it for each admin controller (admin/languages, admin/countries) or can it be assumed that since I'm using the same `before_action` for each controller that it should function? Or should this functionality even just be broken out into the BaseController to run on each route (i'm sure there must be a way to do that)

Reply:
@kolyaventuri I would recommend testing it on each admin route to make sure you didn't miss any.  A downside to this is that it bloats your test suite. Security is pretty important though, so it might be worth the bloat.